lib/Doctrine/DBAL/Portability/Statement.php

```diff
@@ -26,7 +26,7 @@ class Statement implements IteratorAggregate, DriverStatement
     /** @var DriverStatement|ResultStatement */
     private $stmt;
 
-    /** @var int */
+    /** @var int|null */
```

Comment:
This variable [is expected to be nullable](https://github.com/doctrine/dbal/blob/b6e3aa3555e81d3168b4ba96db62ff3f31b93690/lib/Doctrine/DBAL/Portability/Statement.php#L121), however `array_change_key_case()` below expects it to be `int`, never `null`. Hence [another `assert()` below](https://github.com/doctrine/dbal/pull/3801/files#diff-87c7f450dc8a06beead2da33fd34b210R179).

Reply:
This also seems to be the case for 

https://github.com/doctrine/dbal/blob/88f7feae9a74a1f804956470e75284804944d240/lib/Doctrine/DBAL/Portability/Connection.php#L38-L39

Reply:
Indeed, it's null by default (not set in a constructor). And the getter reflects this. Fixed in ff9831475dca77af2e150efd8ad473ad4922b7ce.